### PR TITLE
Default, required, vector tweaks

### PR DESCRIPTION
**Pull request by fwkoch**
- vectors now are only floats
- required default is now `True`
- default values are validated but not coerced, instead the coerced value is saved to the backend on first `get`
  - also, some (maybe?) better handling of `None` vs `undefined` - undefined should be internal only, anything that is set to undefined should be `None` on `get`.

### CODE REVIEW

File: properties/base.py [hunk 1]
```diff
@@ -182,7 +185,10 @@ def _notify(self, change):
 
     def _set(self, name, value):
         self._notify(dict(name=name, value=value, mode='validate'))
-        self._backend[name] = value
+        if value is basic.undefined and name in self._backend:
```
Comment (rowanc1) [inline properties/base.py]: we may want to implement `del` on the property that `_set`s to `undefined`

File: properties/base.py [hunk 2]
```diff
@@ -429,6 +432,46 @@ def name(self, value):
...
+                    prop.validate(None, value())
+                else:
+                    prop.validate(None, value)
+                self._default = value
+                return
+            except Exception:
+                continue
+        raise AssertionError('Invalid default for Union property')
+
+    def _unused_default_warning(self):
```
Comment (rowanc1) [inline properties/base.py]: should this more to `Property`?

**Comment by rowanc1** [comment]
I realize that a lot of the changes were due to linting updates.

**Comment by fwkoch** [comment]
Yes, it all got crammed into one PR, rather than incrementally merged to dev. Smaller PRs in the future.

**Pull request merged**